### PR TITLE
verilator: 5.002 -> 5.006

### PR DESCRIPTION
**Pull request by SharzyL**
###### Description of changes

Changelog: 
- https://verilator.org/guide/latest/changes.html#verilator-5-006-2022-01-22
- https://verilator.org/guide/latest/changes.html#verilator-5-004-2022-12-14

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 214958` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>verilator</li>
  </ul>
</details>

File: pkgs/applications/science/electronics/verilator/default.nix [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
 
-  preCheck = ''
-    patchShebangs \
-      src/flexfix \
-      src/vlcovgen \
-      bin/verilator \
-      bin/verilator_coverage \
-      test_regress/driver.pl \
-      test_regress/t/*.pl
+  preBuild = ''
```
Comment (NickCao) [inline pkgs/applications/science/electronics/verilator/default.nix:27]: postPatch

File: pkgs/applications/science/electronics/verilator/default.nix [hunk 2]
```diff
@@ -1,37 +1,31 @@
...
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-RNoKAEF7zl+WqqbxGP/VvdQqQP8VI3hoQku3b/g0XpU=";
+    hash = "sha256-PA8hbE6XECapuaO5YcgEodOoxSDqpMucdijJBBb7fZg=";
   };
 
   enableParallelBuilding = true;
   buildInputs = [ perl ];
-  nativeBuildInputs = [ flex bison python3 autoconf ];
+  nativeBuildInputs = [ flex bison python3 autoconf help2man git ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/verilator/default.nix:19]: Why do we need git?

File: pkgs/applications/science/electronics/verilator/default.nix (hunk 1 above)
Comment (SharzyL) [reply pkgs/applications/science/electronics/verilator/default.nix]: fixed

File: pkgs/applications/science/electronics/verilator/default.nix (hunk 2 above)
Comment (SharzyL) [reply pkgs/applications/science/electronics/verilator/default.nix]: Fixed. `git` is used to detect version, but seems it fallbacks safely when no git is found.

**Review by NickCao — APPROVED** [review]

**Pull request merged**